core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsIT.java

```diff
@@ -139,6 +138,17 @@ protected int numberOfReplicas() {
         return 1;
     }
 
+    @Override
+    protected void beforeIndexDeletion() {
+        // some test may leave operations in flight
```

Comment:
I think it will be good to explain here that unresponsive disruption swallows request by design, meaning that some operations will never be marked as finished.